tf/edgeml/graph/bonsai.py

```diff
@@ -28,9 +27,9 @@ def __init__(self, numClasses, dataDimension, projectionDimension,
         while doing testing/inference
         numClasses will be reset to 1 in binary case
         '''
-
         self.dataDimension = dataDimension
         self.projectionDimension = projectionDimension
+        self.isRegression = isRegression
```

Comment:
if isRegression is true, throw a warning (if numClasses != 1) and re-assign number of classes to 1

Reply:
I'm already asserting that the number of classes for regression needs to be 1.

Reply:
Both are fine. @metastableB , should it be a warning or assert?

Reply:
Warning - if multi dimensional regression is supported by the code out of the box (this does not mean we support it). Someone might want to hack around the code to perform multi-dim regression and an assert would imply that we don't support it at all. Even though we never designed for it, it might be supported as a consequence of how we coded/formulated.

Assert - if multi-dimensional regression does not make any sense.